test/python/circuit/test_controlled_gate.py

```diff
@@ -290,12 +291,139 @@ def test_multi_control_u1(self):
                 self.log.info(info)
                 self.assertTrue(matrix_equal(target, decomp, ignore_phase=True))
 
-    def test_rotation_gates(self):
-        """Test controlled rotation gates"""
+    @data(1, 2, 3, 4)
+    def test_multi_controlled_u1_matrix(self, num_controls):
+        """Test the matrix representation of the multi-controlled CU1 gate."""
+
+        # registers for the circuit
+        q_controls = QuantumRegister(num_controls)
+        q_target = QuantumRegister(1)
+
+        # iterate over all possible combinations of control qubits
+        for ctrl_state in range(2**num_controls):
+            bitstr = bin(ctrl_state)[2:].zfill(num_controls)[::-1]
+            lam = 0.3165354 * pi
+            qc = QuantumCircuit(q_controls, q_target)
+            for idx, bit in enumerate(bitstr):
+                if bit == '0':
+                    qc.x(q_controls[idx])
+
+            qc.mcu1(lam, q_controls, q_target[0])
+
+            # for idx in subset:
+            for idx, bit in enumerate(bitstr):
+                if bit == '0':
+                    qc.x(q_controls[idx])
+
+            backend = BasicAer.get_backend('unitary_simulator')
+            simulated = execute(qc, backend).result().get_unitary(qc)
+
+            base = U1Gate(lam).to_matrix()
+            expected = _compute_control_matrix(base, num_controls, ctrl_state=ctrl_state)
+            self.assertTrue(matrix_equal(simulated, expected))
```

Comment:
because this assert is in a for-loop, I think `subtest` should be useful here, to avoid blocking the testing if a case fails https://docs.python.org/3/library/unittest.html#distinguishing-test-iterations-using-subtests

Reply:
Agreed, I updated the files!